core/src/main/scala/kafka/controller/PartitionStateMachine.scala

```diff
@@ -283,7 +283,7 @@ class PartitionStateMachine(controller: KafkaController, stateChangeLogger: Stat
     val topicAndPartition = TopicAndPartition(topic, partition)
     val stateChangeLog = stateChangeLogger.withControllerEpoch(controller.epoch)
     // handle leader election for the partitions whose leader is no longer alive
-    stateChangeLog.trace("Started leader election for partition $topicPartition")
+    stateChangeLog.trace(s"Started leader election for partition $topicAndPartition")
```

Comment:
Looks like a trivial refactor to have this method accept `TopicAndPartition` by the way. You could even carry it through to `handleStateChange`. Not required for this patch though.

Reply:
Good suggestion, but better wait for Onur's Controller changes to land first.